src/main/java/org/opentripplanner/routing/algorithm/raptor/transit/constrainedtransfer/TransferIndexGenerator.java

```diff
@@ -0,0 +1,256 @@
+            var c = tx.getTransferConstraint();
+            // Only add transfers witch have an effect on the Raptor routing here.
+            // Some transfers only have the priority set, and that is used in optimized-
+            // transfers, but not in Raptor.
+            if (!c.useInRaptorRouting()) { continue; }
+
+            findTPoints(tx.getFrom(), ALIGHT).stream()
+                    .filter(TPoint::canAlight)
+                    .forEachOrdered(fromPoint -> {
+                        for (var toPoint : findTPoints(tx.getTo(), BOARD)) {
+                            if (toPoint.canBoard() && !fromPoint.equals(toPoint)) {
+                                fromPoint.addTransferConstraints(tx, toPoint);
+                            }
+                        }
+                    });
+        }
+        sortAllTransfersByRanking();
+    }
+
+    private void sortAllTransfersByRanking() {
+        for (var patterns : patternsByRoute.values()) {
+            for (var pattern : patterns) {
+                pattern.sortConstrainedTransfers();
+            }
+        }
+    }
+
+    private void setupPatternByTripIndex(Collection<TripPatternWithRaptorStopIndexes> tripPatterns) {
+        for (TripPatternWithRaptorStopIndexes pattern : tripPatterns) {
+            TripPattern tripPattern = pattern.getPattern();
+
+            patternsByRoute
+                    .computeIfAbsent(tripPattern.getRoute(), t -> new ArrayList<>())
+                    .add(pattern);
+
+            for (Trip trip : tripPattern.getTrips()) {
+                patternsByTrip.computeIfAbsent(trip, t -> new ArrayList<>()).add(pattern);
+            }
+
+            for (StopLocation stop : tripPattern.getStops()) {
+                patternsByStop.computeIfAbsent(stop, t -> new ArrayList<>()).add(pattern);
+                Station station = stop.getParentStation();
+                if (station != null) {
+                    patternsByStation.computeIfAbsent(station, t -> new ArrayList<>()).add(pattern);
+                }
+            }
+        }
+    }
+
+    private Collection<TPoint> findTPoints(TransferPoint txPoint, boolean boarding) {
+        if (txPoint.isStationTransferPoint()) {
+            return findTPoints(txPoint.asStationTransferPoint());
+        }
+        else if (txPoint.isStopTransferPoint()) {
+            return findTPoints(txPoint.asStopTransferPoint());
+        }
+        else if (txPoint.isRouteStationTransferPoint()) {
+            return findTPoint(txPoint.asRouteStationTransferPoint(), boarding);
+        }
+        else if (txPoint.isRouteStopTransferPoint()) {
+            return findTPoint(txPoint.asRouteStopTransferPoint(), boarding);
+        }
+        else {
+            return findTPoints(txPoint.asTripTransferPoint());
+        }
+    }
+
+    private List<TPoint> findTPoints(StationTransferPoint point) {
+        var station = point.getStation();
+        var patterns = patternsByStation.get(station);
```

Comment:
```suggestion
        var patterns = patternsByStop.getOrDefault(stop, Collections.emptyList());
```

Reply:
This was needed to make this work for the Boston GTFS feed: https://cdn.mbta.com/MBTA_GTFS.zip

Reply:
This is fixed in 4950704.